src/analyses/mutexAnalysis.ml

```diff
@@ -15,7 +15,32 @@ module Spec =
 struct
   module Arg =
   struct
-    module D = Lockset
+    module Count = Lattice.Reverse(
+        Lattice.Chain (struct
+          let n () = 5
+          let names x = if x = (n () - 1) then "top" else Format.asprintf "%d" x
+        end))
+
+    module Multiplicity = struct
+      include MapDomain.MapBot_LiftTop (ValueDomain.Addr) (Count)
+      let increment v x =
+        let current = find v x in
+        if current = (4) then
+          x
+        else
+          add v (current + 1) x
+
+      let decrement v x =
+        let current = find v x in
+        if current = 0 then
+          (x, true)
+        else
+          (add v (current - 1) x, current-1 = 0)
+    end
+    module D = struct include Lattice.Prod(Lockset)(Multiplicity)
```

Comment:
Yes, and I thought about it for a while. The idea why I decided against it for now is that it will only be a small minority of mutexes that are recursive, so I want to avoid additional overhead in as many places as possible.
Also, keeping the representation that is needed by other analysis (lockset without multiplicities) around might be advantageous over having to construct it all the time.